tee-worker/omni-executor/pumpx/src/pumpx_api/types.rs

```diff
@@ -0,0 +1,101 @@
+use parity_scale_codec::{Codec, Decode, Encode};
+use serde::{Deserialize, Serialize};
+use serde_repr::{Deserialize_repr, Serialize_repr};
+
+pub type UserConnectResponse = ApiResponse<ConnectedUser>;
+
+pub type MarketOrderUnsignedTxResponse = ApiResponse<MarketOrderUnsignedTx>;
+
+pub type MarketOrderTxResponse = ApiResponse<TxData>;
+
+#[derive(Deserialize_repr, Serialize_repr)]
+#[allow(clippy::upper_case_acronyms)]
+#[repr(u8)]
+pub enum ChainId {
+	EVM = 1,
+	Solana = 2,
+}
+
+#[derive(Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ConnectUser {
+	pub email: String,
+	pub invite_code: Option<String>,
+	pub google_code: Option<String>,
+}
+
+#[derive(Deserialize, Serialize, Decode, Encode, PartialEq, Eq, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct ConnectedUser {
+	pub user_id: String,
+	pub google_auth_check: bool,
+}
+
+#[derive(Deserialize, Serialize, Encode, Decode, PartialEq, Eq, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct ApiResponse<T: Codec> {
+	code: u32,
+	message: String,
+	data: T,
+}
```

Comment:
I like the abstraction of this 👍 We just need to double-check later if every response follows it

Reply:
All of the responses in the docs I've seen have this shape, hopefully it's consistent